pkg/kn/commands/service/apply.go

```diff
@@ -75,15 +78,28 @@ func NewServiceApplyCommand(p *commands.KnParams) *cobra.Command {
 			if err != nil {
 				return err
 			}
+			if trafficFlags.Changed(cmd) {
+				service.Spec.Traffic, err = traffic.Compute(cmd, service.Spec.Traffic, &trafficFlags, service.Name)
+				if err != nil {
+					return err
+				}
+			}
 
 			client, err := p.NewServingClient(namespace)
 			if err != nil {
 				return err
 			}
 
-			waitDoing, waitVerb, err := examineServiceForApply(cmd, client, service.Name)
-			if err != nil {
+			waitDoing, waitVerb := "Applying", "applied"
+			if isCreate, err := examineServiceForApply(cmd, client, service.Name); err != nil {
```

Comment:
Why not is,ply this if/else/if with simply:

```go
err := examineServiceForApply(cmd, client, service.Name)
if err != nil {
  return err
}

// do as in the else / if since the ifCreate condition applies to both parts of the statement
// and just add a `if !isCreate ...` guard and return or do what's needed in that case in a helper function